File: LR3/app/src/main/java/com/example/lr3/types/users/UserType.java
package com.example.lr3.types.users;

import com.example.lr3.comparator.Comparator;

import java.io.IOException;
import java.io.InputStreamReader;

public interface UserType {

    String typeName();

    Object create() throws InstantiationException;

    Object clone(Object object) throws InstantiationException;

    Object readValue(InputStreamReader in) throws IOException;

    Object parseValue(String ss) throws InstantiationException;

    Comparator getTypeComparator();

    String toString(Object object);
}

File: LR2/src/main/kotlin/types/users/UserType.kt
package types.users

import comparator.Comparator
import java.io.IOException
import java.io.InputStreamReader

interface UserType {
    fun typeName(): String

    @Throws(InstantiationException::class)
    fun create(): Any

    @Throws(InstantiationException::class)
    fun clone(`object`: Any): Any

    @Throws(IOException::class)
    fun readValue(`in`: InputStreamReader?): Any

    @Throws(InstantiationException::class)
    fun parseValue(ss: String): Any?
    val typeComparator: Comparator
    fun toString(`object`: Any): String
}
